src/NzbDrone.Core/HealthCheck/Checks/RemotePathMappingCheck.cs

```diff
@@ -0,0 +1,112 @@
+using NzbDrone.Core.MediaFiles.Events;
+using NzbDrone.Core.RemotePathMappings;
+using NzbDrone.Core.ThingiProvider.Events;
+
+namespace NzbDrone.Core.HealthCheck.Checks
+{
+    [CheckOn(typeof(ProviderAddedEvent<IDownloadClient>))]
+    [CheckOn(typeof(ProviderUpdatedEvent<IDownloadClient>))]
+    [CheckOn(typeof(ProviderDeletedEvent<IDownloadClient>))]
+    [CheckOn(typeof(ModelEvent<RemotePathMapping>))]
+    [CheckOn(typeof(TrackImportedEvent), CheckOnCondition.FailedOnly)]
+    [CheckOn(typeof(TrackImportFailedEvent), CheckOnCondition.SuccessfulOnly)]
+    public class RemotePathMappingCheck : HealthCheckBase, IProvideHealthCheckWithMessage
+    {
+        private readonly IDiskProvider _diskProvider;
+        private readonly IProvideDownloadClient _downloadClientProvider;
+        private readonly Logger _logger;
+
+        public RemotePathMappingCheck(IDiskProvider diskProvider,
+                                      IProvideDownloadClient downloadClientProvider,
+                                      Logger logger)
+        {
+            _diskProvider = diskProvider;
+            _downloadClientProvider = downloadClientProvider;
+            _logger = logger;
+        }
+
+        public override HealthCheck Check()
+        {
+            var clients = _downloadClientProvider.GetDownloadClients();
+
+            foreach (var client in clients)
+            {
+                var folders = client.GetStatus().OutputRootFolders;
+                if (folders != null)
+                {
+                    foreach (var folder in folders)
+                    {
+                        if (!_diskProvider.FolderExists(folder.FullPath))
+                        {
+                            if (OsInfo.IsDocker)
+                            {
+                                return new HealthCheck(GetType(), HealthCheckResult.Error, $"You are using docker; download client {client.Definition.Name} places downloads in {folder.FullPath} but this directory does not appear to exist inside the container.  Review your remote path mappings and container volume settings.", "#docker-bad-remote-path-mapping");
+                            }
+                            else if (!client.GetStatus().IsLocalhost)
+                            {
+                                return new HealthCheck(GetType(), HealthCheckResult.Error, $"Remote download client {client.Definition.Name} places downloads in {folder.FullPath} but this directory does not appear to exist.  Likely missing or incorrect remote path mapping.", "#bad-remote-path-mapping");
+                            }
+                            else
+                            {
+                                return new HealthCheck(GetType(), HealthCheckResult.Error, $"Download client {client.Definition.Name} places downloads in {folder.FullPath} but Lidarr cannot see this directory.  Check that the user running Lidarr has the necessary permissions.", "#permissions-error");
```

Comment:
Would this case always be permissions? If folder doesn't exist and client is local host, does that necessarily mean that its a permissions issue?

Reply:
I'm sure there would be some case where it's not a permissions issue :) Maybe I should make the language a bit looser?  It could be for example that the client hasn't created the category folder yet.